app/assets/v2/js/cart-data.js

```diff
@@ -52,6 +49,10 @@ class CartData {
   }
 
   static addToCart(grantData, no_report) {
+    // Enforce that grant ID is a number, since backend returns it as a number
+    grantData.grant_id = Number(grantData.grant_id);
```

Comment:
This is breaking `removeIdFromCart` - could we make sure we're comparing Number()'s against Number()'s in removeIdFromCart's filter please? :)